scripts/dev/ci_timing_summary.py

```
@@ -0,0 +1,185 @@
+    for step in summary.slowest_steps:
+        lines.append(
+            "| "
+            f"{step.name} | {_format_seconds(step.duration_seconds)} | "
+            f"{step.started_at} | {step.completed_at} |",
+        )
+    return "\n".join(lines)
+
+
+def _load_payload_from_gh(run_id: str) -> dict[str, Any]:
+    """Fetch a workflow run payload through the GitHub CLI."""
+    output = subprocess.check_output(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)